### ISSUE TITLE
The size of tensor a (128) must match the size of tensor b (88) at non-singleton dimension 3

### ISSUE THREAD

**Issue by exinnocommon** [tgohblio/InstantID-MultiControlNet#1]
Thank you for creating such an amazing Cog model.

When running with docker the following error occurs.

Environment:
RTX 3090 Ti
Ubuntu 22.04
Driver Version: 525.147.05   CUDA Version: 12.0

Install command:
docker run -d -p 5011:5000 --name instant-id-multicontrolnet2 --gpus device=0 --restart=always r8.im/tgohblio/instant-id-multicontrolnet@sha256:52039b70bd022865062c51eb77f36103e3605e540e1f605652e205c4a7eadd48

Error logs:
{"prediction_id": null, "logger": "cog.server.runner", "timestamp": "2024-04-18T15:33:40.062248Z", "severity": "INFO", "message": "starting prediction"}
  0%|          | 0/4 [00:00<?, ?it/s]
0%|          | 0/4 [00:00<?, ?it/s]
Traceback (most recent call last):
File "/root/.pyenv/versions/3.11.8/lib/python3.11/site-packages/cog/server/worker.py", line 217, in _predict
result = predict(**payload)
^^^^^^^^^^^^^^^^^^
File "/root/.pyenv/versions/3.11.8/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^
File "/src/cog/predict.py", line 550, in predict
image = self.pipe(
^^^^^^^^^^
File "/root/.pyenv/versions/3.11.8/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^
File "/src/pipeline_stable_diffusion_xl_instantid_full.py", line 1129, in __call__
down_block_res_samples = [
^
File "/src/pipeline_stable_diffusion_xl_instantid_full.py", line 1130, in <listcomp>
down_block_res_sample * mask_weight
~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
RuntimeError: The size of tensor a (128) must match the size of tensor b (88) at non-singleton dimension 3
{"error": "The size of tensor a (128) must match the size of tensor b (88) at non-singleton dimension 3", "prediction_id": null, "logger": "cog.server.runner", "timestamp": "2024-04-18T15:33:43.653537Z", "severity": "INFO", "message": "prediction failed"}

**Comment by exinnocommon** [comment]
It seems to be an issue with the resolution of the original photo. The problem seems to occur when the aspect ratio is different.

**Comment by platform-kit** [comment]
@exinnocommon @tgohblio I'm getting this error when running on replicate, even when the input image's dimensions match the resolution I specify.

**Comment by tgohblio** [comment]
Thanks, let me have a look

**Comment by tgohblio** [comment]
@exinnocommon @platform-kit Seems like the aspect ratio of the final output image has to match the input face image, or the pose image. So custom width and height settings are removed. If you need it, please use another image resize tool.
 
Fixes are in this commit: [c3418d89a9cffb3be3eee1558d096279814556e2](https://github.com/tgohblio/InstantID-MultiControlNet/commit/c3418d89a9cffb3be3eee1558d096279814556e2)
